### ISSUE TITLE
No performance improvements found with Graal EE and the given flags

### ISSUE THREAD

**Issue by KB-Q** [brucethemoose/Minecraft-Performance-Flags-Benchmarks#12]
I followed this guide, installed GraalVM EE (22.1.0) and pasted the flags from the different sections onto my PolyMC 1.19 instance. I'm sad to say I didn't see any performance improvements at all with these (infact I even had freezes when loading a lot of chunks quickly, but these were infrequent). Before this, I used Adoptium's JDK with Aikar's G1GC flags, and I got around 50-60 fps with constant lag spikes when generating new chunks. This situation is pretty much the same with GraalVM and these new flags.

These are the flags I used:
`-XX:+UnlockExperimentalVMOptions -XX:+UnlockDiagnosticVMOptions -XX:+AlwaysPreTouch -XX:+DisableExplicitGC -XX:+UseNUMA -XX:AllocatePrefetchStyle=3 -XX:NmethodSweepActivity=1 -XX:ReservedCodeCacheSize=400M -XX:NonNMethodCodeHeapSize=12M -XX:ProfiledCodeHeapSize=194M -XX:NonProfiledCodeHeapSize=194M -XX:-DontCompileHugeMethods -XX:MaxNodeLimit=240000 -XX:NodeLimitFudgeFactor=8000 -XX:+UseVectorCmov -XX:+PerfDisableSharedMem -XX:+UseFastUnorderedTimeStamps -XX:+UseCriticalJavaThreadPriority -XX:+EnableJVMCIProduct -XX:+UseJVMCICompiler -XX:+EagerJVMCI -Dgraal.TuneInlinerExploration=1 -Dgraal.CompilerConfiguration=enterprise -Dgraal.UsePriorityInlining=true -Dgraal.Vectorization=true -Dgraal.OptDuplication=true -Dgraal.DetectInvertedLoopsAsCounted=true -Dgraal.LoopInversion=true -Dgraal.VectorizeHashes=true -Dgraal.EnterprisePartialUnroll=true -Dgraal.VectorizeSIMD=true -Dgraal.StripMineNonCountedLoops=true -Dgraal.SpeculativeGuardMovement=true -Dgraal.InfeasiblePathCorrelation=true -Dgraal.LoopRotation=true -XX:+UseG1GC -XX:MaxGCPauseMillis=37 -XX:+PerfDisableSharedMem -XX:G1HeapRegionSize=16M -XX:G1NewSizePercent=23 -XX:G1ReservePercent=20 -XX:SurvivorRatio=32 -XX:G1MixedGCCountTarget=3 -XX:G1HeapWastePercent=20 -XX:InitiatingHeapOccupancyPercent=10 -XX:G1RSetUpdatingPauseTimePercent=0 -XX:MaxTenuringThreshold=1 -XX:G1SATBBufferEnqueueingThresholdPercent=30 -XX:G1ConcMarkStepDurationMillis=5.0 -XX:G1ConcRSHotCardLimit=16 -XX:G1ConcRefinementServiceIntervalMillis=150 -XX:GCTimeRatio=99 -XX:+UseLargePages -XX:LargePageSizeInBytes=2m`

I also use a lot of performance mods, given in this list: https://paste.ee/p/JaFLl

My system specs:
Intel i7-8565U CPU, with Integrated Graphics
Allocated memory to Minecraft: 4 GB

Any idea what should I do in this case? Am I just doomed to have constant lag spikes because of my system, or am I doing something wrong? Please help me with this. Thanks :)

**Comment by brucethemoose** [comment]
Make your memory allocation static, as described in the guide.

Also try running `/sparkc gcmonitor` to see if your lag spikes are coming from garbage collection or something else.



Unfortunately, running GraalVM EE has 2 caveats:

- It doesn't help client fps at all, at least on my machine. It basically only speeds up the internal minecraft server in SP (though it does measurable speed up chunkgen)
- It increases background CPU usage. This could be a problem on your 8565U CPU.

Try running regular OpenJDK with the flags, and see if its any better.

Also be sure to try the process priority and SpecialK tweaks if you haven't already.

**Comment by KB-Q** [comment]
> Make your memory allocation static, as described in the guide.
You mean set both the minimum and maximum to the same value?

> Try running regular OpenJDK with the flags, and see if its any better.
In that case should I remove any of the flags; especially the `-Dgraal` ones?

**Comment by brucethemoose** [comment]
> > Make your memory allocation static, as described in the guide.
> 
> You mean set both the minimum and maximum to the same value?
> 
> > Try running regular OpenJDK with the flags, and see if its any better.
> 
> In that case should I remove any of the flags; especially the `-Dgraal` ones?

Yeah, use the set for regular openjdk instead of the dgrall ones.

**Comment by abrasmag3** [comment]
> - It doesn't help the client fps at all. It basically only speeds up the internal server...

Does another JRE help more with client FPS, but lack in server FPS? That could be an important thing to note for instances that are designed to be used on servers.
